libvast/vast/chunk.hpp

```diff
@@ -77,21 +78,14 @@ class chunk : public caf::ref_counted {
   static chunk_ptr make(std::vector<Byte>&& xs) {
     static_assert(sizeof(Byte) == 1);
     VAST_ASSERT(std::size(xs) != 0);
-    auto ys = std::make_shared<std::vector<Byte>>(std::move(xs));
-    auto deleter = [=]() mutable { ys.reset(); };
-    return make(ys->size(), ys->data(), deleter);
+    auto shared = std::make_shared<std::vector<Byte>>(std::move(xs));
+    return make(shared->size(), shared->data(), detail::keeper{shared});
   }
 
   // Construct a chunk by copying a range of bytes.
-  template <typename Byte>
-  static chunk_ptr copy(span<Byte> span) {
-    static_assert(sizeof(Byte) == 1);
-    auto size = span.size();
-    VAST_ASSERT(size > 0);
-    auto chunk = make(size);
-    if (chunk)
-      std::memcpy(chunk->data_, span.data(), size);
-    return chunk;
+  template <typename Byte, size_t Extent>
+  static chunk_ptr copy(span<Byte, Extent> span) {
+    return make(std::vector(span.begin(), span.end()));
```

Comment:
2 Flies with one stone?

Reply:
Yes, unless that's a problem. I want to do a second cleanup pass on the chunk API sometime soon anyways, so I could move this change in that upcoming PR as well.